crates/core_simd/src/vector.rs

```diff
@@ -265,38 +290,76 @@ where
     /// let enable = enable & idxs.lanes_lt(Simd::splat(vec.len()));
     ///
     /// // We have masked the OOB lane, so it's safe to scatter now.
-    /// unsafe { vals.scatter_select_unchecked(&mut vec, enable, idxs); }
+    /// unsafe { vals.scatter_select_unchecked(vec.as_mut_ptr(), enable, idxs); }
     /// // index 0's second write is masked, thus was omitted.
     /// assert_eq!(vec, vec![-41, 11, 12, 82, 14, 15, 16, 17, 18]);
     /// ```
     /// [undefined behavior]: https://doc.rust-lang.org/reference/behavior-considered-undefined.html
     #[inline]
     pub unsafe fn scatter_select_unchecked(
         self,
-        slice: &mut [T],
+        ptr: *mut T,
         enable: Mask<isize, LANES>,
         idxs: Simd<usize, LANES>,
     ) {
-        // SAFETY: This block works with *mut T derived from &mut 'a [T],
+        // SAFETY: This block works with a *mut T that may be from &mut 'a [T],
         // which means it is delicate in Rust's borrowing model, circa 2021:
-        // &mut 'a [T] asserts uniqueness, so deriving &'a [T] invalidates live *mut Ts!
-        // Even though this block is largely safe methods, it must be exactly this way
-        // to prevent invalidating the raw ptrs while they're live.
+        // &mut 'a [T] asserts uniqueness, so deriving &'a [T] invalidates live *mut T!
```

Comment:
I don't quite understand this comment.

`deriving &'a [T]` *from the original `&mut` that sources this `*mut` would invalidate the `*mut`, but we cannot even do that here since we do not have access to that original `&mut`. As long as we work with raw pointers only, no aliasing restrictions arise.

I am probably suffering from a lack of imagination here -- what is the alternative version of this code that would be wrong?

Reply:
This is a leftover comment from the code on main in fact accepting `&mut [T]`.